tests/integration/contracts/blockifier_contracts/feature_contracts/cairo0/test_proxy_account.cairo

```diff
@@ -0,0 +1,98 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.memcpy import memcpy
+
+from starkware.cairo.common.math import assert_nn
+from starkware.starknet.common.syscalls import (
+    get_tx_info
+)
+
+struct Call {
+    to: felt,
+    selector: felt,
+    calldata_len: felt,
+    calldata: felt*,
+}
+
+struct CallArray {
+    to: felt,
+    selector: felt,
+    data_offset: felt,
+    data_len: felt,
+}
+
+@external
+func __validate__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    ecdsa_ptr: SignatureBuiltin*,
+    range_check_ptr
+} (
+    call_array_len: felt,
+    call_array: CallArray*,
+    calldata_len: felt,
+    calldata: felt*
+) {
+    return ();
+}
+
+@external
+@raw_output
+func __execute__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    ecdsa_ptr: SignatureBuiltin*,
+    range_check_ptr
+} (
+    call_array_len: felt,
+    call_array: CallArray*,
+    calldata_len: felt,
+    calldata: felt*
+) -> (
+    retdata_size: felt, retdata: felt*
+) {
+    alloc_locals;
+    let (retdata: felt*) = alloc();
+
+    return (retdata_size=0, retdata=retdata);
+}
+
+@external
+func __validate_declare__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    ecdsa_ptr: SignatureBuiltin*,
+    range_check_ptr
+} (
+    class_hash: felt
+) {
+    return ();
+}
+
+@raw_input
+@external
+func __validate_deploy__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    ecdsa_ptr: SignatureBuiltin*,
+    range_check_ptr
+} (selector: felt, calldata_size: felt, calldata: felt*) {
+    alloc_locals;
+    // get the tx info
+    let (tx_info) = get_tx_info();
+    with_attr error_message("assertion failed") {
+        assert_nn(tx_info.signature_len - 2);
+    }
+
+    return ();
+}
```

Comment:
Is this still needed? If we don't need all of the unusual things here (e.g. @raw_input) in order to test this (as a regression test) then I would suggest removing it or at least adding some documentation about it. Otherwise I can see it adding a lot of confusion in the future.

Reply:
Resolved by https://github.com/keep-starknet-strange/snos/pull/431/commits/c4726a7c44e2a64d7156aeaea7075d52c14ee601